src/views/v2/list/top-bar/datepicker.php

```diff
@@ -11,59 +11,30 @@
+ * @var string $selected_start_datetime    The selected start date and time in the localized `Y-m-d` format.
+ * @var string $selected_start_date_mobile The formatted date that will show in the datepicker mobile version.
+ * @var string $selected_start_date_label  The label of the datepicker interval start.
+ * @var string $selected_end_datetime      The selected end date and time in the localized `Y-m-d` format.
+ * @var string $selected_end_date_mobile   The formatted date that will show in the datepicker mobile version.
+ * @var string $selected_end_date_label    The label of the datepicker interval end.
+ * @var string $datepicker_date            The datepicker selected date, in the `Y-m-d` format.
  */
-use Tribe__Date_Utils as Dates;
 
-/**
- * @todo @bordoni @lucatume Abstract the handling of these dates into a method/function
- */
-$default_start_date        = $now;
-$selected_start_date_value = $this->get( [ 'bar', 'date' ], $default_start_date );
-if ( empty( $selected_start_date_value ) ) {
-	$first_event = $this->get( 'view' )->get_repository()->first();
-
-	if ( $first_event instanceof WP_Post ) {
-		$selected_start_date_value = $first_event->dates->start->format( Dates::DBDATETIMEFORMAT );
-	} else {
-		$selected_start_date_value = $default_start_date;
-	}
-}
-
-$selected_start_datetime = strtotime( $selected_start_date_value );
-$is_now                  = Dates::build_date_object( $selected_start_datetime )->format( Dates::DBDATEFORMAT ) === Dates::build_date_object( $default_start_date )->format( Dates::DBDATEFORMAT );
-
-$selected_end_date_value = $today;
-$last_event              = $this->get( 'view' )->get_repository()->last();
-
-if ( $last_event instanceof WP_Post ) {
-	$selected_end_date_value = $last_event->dates->start->format( Dates::DBDATEFORMAT );
-}
-$selected_end_datetime = strtotime( $selected_end_date_value );
-
-$selected_start_date_label_format = tribe_get_date_format( date( 'Y' ) !== date( 'Y', $selected_start_datetime ) );
-$selected_end_date_label_format   = tribe_get_date_format( date( 'Y' ) !== date( 'Y', $selected_end_datetime ) );
-
-$selected_start_date_label = date_i18n( $selected_start_date_label_format, $selected_start_datetime );
-$selected_end_date_label   = date_i18n( $selected_end_date_label_format, $selected_end_datetime );
-
-$selected_start_date_mobile = Dates::build_date_object( $selected_start_datetime )->format( $date_formats->compact );
-$selected_end_date_mobile   = Dates::build_date_object( $selected_end_datetime )->format( $date_formats->compact );
-
-$datepicker_date = Dates::build_date_object( $selected_start_date_value )->format( $date_formats->compact );
 ?>
 <div class="tribe-events-c-top-bar__datepicker">
 	<button
 		class="tribe-common-h3 tribe-common-h--alt tribe-events-c-top-bar__datepicker-button"
 		data-js="tribe-events-top-bar-datepicker-button"
 	>
-		<?php if ( $is_now ) : ?>
-			<?php esc_html_e( 'Now', 'the-events-calendar' ); ?>
+		<?php if ( $show_now ) : ?>
```

Comment:
could we add a test for this in the partials test?

Reply:
I've added snapshot tests for this and removed two todos (for me) I found while there.